client/lib/analytics/index.js

```diff
@@ -781,16 +781,19 @@ const analytics = {
 		},
 	},
 
-	identifyUser: function() {
+	identifyUser: function( { user = _user, newUserId = '', newUserName = '' } = {} ) {
```

Comment:
What if this function had signature `identifyUser( userId, userName )`? That would make the code a lot simpler and remove the dependency on `lib/user`.

And `identifyUser` is always called with valid `userId` and `userName`, so some checks are redundant, too.

Reply:
Thanks @jsnajdr !! I like the idea.

`identifyUser` is also called [here](https://github.com/Automattic/wp-calypso/blob/bf89ed0806d761dbb6a97a769425640c9acc6781/client/lib/analytics/index.js#L200) without any params.

The `user` prop I added for unit test purposes.

An improvement for later?

Reply:
> identifyUser is also called here without any params.

Yes, and the `analytics.initialize` function receives `user` as param. So the `username` and `ID` can be extracted and passed to `identifyUser`.

`analytics.initialize` is called only when the user is valid (i.e., logged in): https://github.com/Automattic/wp-calypso/blob/master/client/boot/common.js#L266

> The user prop I added for unit test purposes.

Unit tests would get simpler: no need to mock `lib/user` -- just pass the testing data as parameters!

> An improvement for later?

If "later" doesn't mean "never", then I agree 😄

Reply:
>  So the username and ID can be extracted and passed to identifyUser.

Ahhh 💡! Thanks for that. I'll fix it. Appreciate your help.